src/EditorFeatures/Core.Wpf/InlineRename/Dashboard/DashboardViewModel.cs

```diff
@@ -65,12 +66,36 @@ private void OnReferenceLocationsChanged(object sender, ImmutableArray<InlineRen
             UpdateSearchText(totalSpansCount, totalFilesCount);
         }
 
+        private void OnIsReplacementTextValidChanged(bool isReplacementTextValid)
```

Comment:
```suggestion
        private void OnIsReplacementTextValidChanged(bool isReplacementTextValid)
        {
            if (_isReplaceTextValid == isReplacementTextValid)
            {
                return;
            }
            
            _isReplacementTextValid = isReplacementTextValid;
            ComputeDefaultRenameFileFlag();
            NotifyPropertyChanged(nameof(AllowFileRename)); // Not sure if this is needed here, but we want to do NPC if the value is going to be computed differently
        }
        
        void ComputeDefaultRenameFileFlag()
        {
            DefaultRenameFileFlag = _isReplacementTextValid && 
                (_session.Option.GetOption(RenameOptions.RenameFile) || AllowFileRename);
        }
```

Reply:
You can then call `ComputeDefaultRenameFileFlag()` in the constructor as well so the logic is the same in both places